tutorials/03_moduly.md

```diff
@@ -0,0 +1,350 @@
+
+```python
+setup(
+    install_requires=['Flask', 'click>=6'],
+)
+```
+
+Upload na PyPI
+--------------
+
+Balíček jde zaregistrovat a nahrát na PyPI. Původně k tomu sloužily příkazy
+`setup.py` `register` a `upload`, ale tyto příkazy používají HTTP, což není
+bezpečné. Prototo je lepší použít program `twine` (instalovatelný přes pip).
+
+Budete potřebovat [účet na PyPI](https://pypi.python.org/pypi?%3Aaction=register_form).
+[účet na testovací PyPI](https://testpypi.python.org/pypi?%3Aaction=register_form)
+a konfigurační soubor `~/.pypirc`:
+
+```ini
+[distutils]
+index-servers=
+    pypi
+    pypitest
+
+[pypi]
+repository = https://pypi.python.org/pypi
+username = <your user name goes here>
+password = <your password goes here>
+
+[pypitest]
+repository = https://testpypi.python.org/pypi
+username = <your user name goes here>
+password = <your password goes here>
+```
+
+Hesla můžete vynechat, pokud je budete chtít pokaždé zadávat.
+
+Registrace projektu a nahrání na testovací PyPI se provádí pomocí:
+
+```bash
+(env)$ twine register -r pypitest dist/<soubor>
+Registering package to https://testpypi.python.org/pypi
+Registering <soubor>
+(env)$ twine upload -r pypitest dist/<soubor>
+Uploading distributions to https://testpypi.python.org/pypi
+Uploading <soubor>
+[================================] 8379/8379 - 00:00:02
+```
+
+Registrace se zdaří jen pokud jméno projektu již není zabrané.
+Po úspěšném nahrání lze nahrát už jen novější verze modulu.
+
+Pro nahrání na opravdovou PyPi stačí vynechat `-r pypitest`.
+
+Další
+-----
+
+ * [instalace datových souborů](https://docs.python.org/3/distutils/setupscript.html#installing-package-data)
+ * [obsáhlá dokumentace](https://packaging.python.org/)
+
+Úkol
+----
+
+Vaším úkolem za 5 bodů je udělat z vašeho dosavadního projektu balíček
+instalovatelný přes pip a nahrát jej na testovací nebo opravdovou PyPI.
+
+Na opravdovou PyPI prosím nahrávejte pouze s rozumným názvem a pokud jde o dílo
+s nějakou open-source licencí.
+
+Pokud svůj kód za žádných okolností nechcete zveřejnit ani na testovací PyPI,
+dejte nám vědět a domluvíme se.
+
+Podmínky (je jich hodně, ale jsou triviální):
+
+ * Váš balíček musí fungovat (viz zadání předchozích úkolů) po instalaci pomocí pipu do "prázdného" virtualenvu.
+ * Musí instalovat potřebné závislosti.
+ * Musí obsahovat rozumný počet classsifiers a voleb pro `setup.py`.
+ * Podpříkaz `sdist` nesmí skončit chybou ani vyvolat varování.
+ * Musí splňovat zde uvedené konvence.
+ * Hlavní skript musí jít spouštět pomocí entry pointu i pomocí `-m`.
+ * Modul musí obsahovat `__init__.py` a logiku importovat z ostatních souborů.
+ * Zabalený modul musí obsahovat soubor s textem licence (`LICENSE`, `COPYING`) \*
+ * `long_description` musí být načten z `README`
+
+\* V případě, že váš kód nechcete šířit pod svobodnou licencí, napište to do souboru vlastní podmínky.
+
+Odevzdáte tagem v0.3 v obvyklém repozitáři (ten můžete klidně přejmenovat podle modulu).
```

Comment:
Pokud to budou přejmenovávat, tak ať to dělají na Githubu přes "Settings". Jinak se může stát že to Github nepropojí a nezařídí redirecty, takže to nové repo nenajdeme.